bids/bids.tf

```diff
@@ -0,0 +1,306 @@
+    },
+    {
+      price = 1.97
+      plan  = "c2.large.arm"
+      name  = ""
+      url   = "https://netboot.gsc.io/hydra-aarch64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.96
+      plan  = "c2.large.arm"
+      name  = "c2.large.arm--armv7l"
+      url   = "https://netboot.gsc.io/hydra-aarch64-linux/netboot.ipxe"
+    },
+
+    {
+      price = 2.0
+      plan  = "m1.xlarge.x86"
+      name  = "m1.xlarge.x86-big--parallel"
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.99
+      plan  = "m1.xlarge.x86"
+      name  = "m1.xlarge.x86-big--parallel"
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.98
+      plan  = "m1.xlarge.x86"
+      name  = ""
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.97
+      plan  = "m1.xlarge.x86"
+      name  = ""
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.96
+      plan  = "m1.xlarge.x86"
+      name  = ""
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.95
+      plan  = "m1.xlarge.x86"
+      name  = ""
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+    {
+      price = 1.94
+      plan  = "m1.xlarge.x86"
+      name  = ""
+      url   = "https://netboot.gsc.io/hydra-x86_64-linux/netboot.ipxe"
+    },
+  ]
+}
+
+locals {
+  named_bids = { for bid in var.bids : "${bid.plan}--${bid.name}--${bid.price}" => bid }
+}
+
+resource "metal_spot_market_request" "request" {
+  for_each      = local.named_bids
+  project_id    = var.project_id
+  max_bid_price = each.value.price
+  facilities    = ["ams1", "sjc1", "dfw2", "nrt1", "ewr1"]
+  devices_min   = 1
+  devices_max   = 1
+
+  instance_parameters {
+    hostname         = each.value.name == "" ? each.value.plan : each.value.name
+    billing_cycle    = "hourly"
+    operating_system = "custom_ipxe"
+    always_pxe       = true
+    plan             = each.value.plan
+    ipxe_script_url  = each.value.url
+    project_ssh_keys = []
+    user_ssh_keys    = []
+    tags             = concat(var.tags, ["hydra"])
+  }
+}
+
+
+resource "metal_device" "reservation" {
+  for_each                = { for id, value in var.reservations : id => value if ! contains(var.reservation_broken, id) }
+  project_id              = var.project_id
+  hostname                = lookup(var.reservation_class_names, each.value.class, each.value.class)
+  billing_cycle           = "hourly"
+  operating_system        = "custom_ipxe"
+  always_pxe              = true
+  plan                    = each.value.class
+  facilities              = [var.reservation_facility_names[each.value.facility]]
+  hardware_reservation_id = each.key
+
+  ipxe_script_url                  = var.reservation_class_urls[each.value.class]
+#  user_data = <<USERDATA
```

Comment:
Does this commented out block still serve some purpose? I guess knowing how to set ntp etc.. is good but do we need that?

Reply:
Pruned :)